src/collective/elasticsearch/patches/__init__.py

```diff
@@ -1,71 +1,74 @@
-from collective.elasticsearch import hook
-from collective.elasticsearch.es import ElasticSearchCatalog
-from plone import api
-
-
-def catalog_object(self, obj, uid=None, idxs=None, update_metadata=1, pghandler=None):
-    if idxs is None:
-        idxs = []
-    es = ElasticSearchCatalog(self)
-    return es.catalog_object(obj, uid, idxs, update_metadata, pghandler)
-
-
-def uncatalog_object(self, uid, obj=None, *args, **kwargs):  # NOQA W1113
-    es = ElasticSearchCatalog(self)
-    return es.uncatalog_object(uid, obj, *args, **kwargs)
+from collective.elasticsearch import interfaces
+from collective.elasticsearch.manager import ElasticSearchManager
+from plone.folder.interfaces import IOrdering
+from Products.CMFCore.indexing import processQueue
+from Products.CMFCore.interfaces import IContentish
+from zope.globalrequest import getRequest
+from zope.interface import alsoProvides
+from zope.interface import noLongerProvides
 
 
 def unrestrictedSearchResults(self, REQUEST=None, **kw):
-    es = ElasticSearchCatalog(self)
-    return es.searchResults(REQUEST, check_perms=False, **kw)
+    manager = ElasticSearchManager()
+    return manager.search_results(REQUEST, check_perms=False, **kw)
 
 
 def safeSearchResults(self, REQUEST=None, **kw):
```

Comment:
Plone's `searchResults` method calls [processQueue](https://github.com/plone/Products.CMFPlone/blob/811621d057c755db97a428c9722702761a59e6c8/Products/CMFPlone/CatalogTool.py#L401). So you need to call it in the patch.

Reply:
Done!